lib/SIL/SILFunctionType.cpp

```diff
@@ -1056,7 +1237,15 @@ static CanSILFunctionType getSILFunctionType(
   SILCoroutineKind coroutineKind = SILCoroutineKind::None;
   SmallVector<SILYieldInfo, 8> yields;
   destructureYieldsForCoroutine(TC, expansionContext, origConstant, constant,
-                                reqtSubs, yields, coroutineKind);
+                                reqtSubs, yields, coroutineKind, subst);
+  
+  // Destructure the result tuple type.
+  SmallVector<SILResultInfo, 8> results;
+  {
+    DestructureResults destructurer(expansionContext, TC, conventions,
+                                    results, subst);
+    destructurer.destructure(origResultType, substFormalResultType);
+  }
```

Comment:
Out of curiosity, why did the order of destructuring need to change here?  Oh, just so that type parameters in parameters would get named first?

Reply:
Yeah, the exact ordering didn't seem to be important, and I was trying to keep the order of introduced generic parameters for substituted types consistent with what I had implemented before.